Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/submission/SubmissionResultPositiveOtherWarningFragmentTest.kt

```diff
@@ -0,0 +1,69 @@
+package de.rki.coronawarnapp.ui.submission
+
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.platform.app.InstrumentationRegistry.getInstrumentation
+import androidx.test.rule.ActivityTestRule
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.ui.main.MainActivity
+import de.rki.coronawarnapp.ui.main.MainFragment
+import org.junit.Assert.assertNotNull
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+
+class SubmissionResultPositiveOtherWarningFragmentTest {
+    @Rule
+    @JvmField
+    //Create ActivityTestRule
+    var activityTestRule: ActivityTestRule<MainActivity?>? = ActivityTestRule(MainActivity::class.java)
+    lateinit var activity: MainActivity
+
+    @Before
+    //Setup activity before the test
+    fun SetUP() {
+        activity = activityTestRule!!.activity!!
+    }
+
+    /**
+     * Positive result test
+     * to be executed in a scenario where test results are positive
+     */
+    @Test
+    fun performPositiveTestResultTest() {
```

Comment:
I´m not sure how this test is supposed to work. I´m at least missing some sort of setup of the test case. Also is this a test for the positive test result case or for the special case of "no diagnosis keys"?

Reply:
This is a unit test which looks for positive test result case. and then checks if there are no keys. I have a feeling I'll have to rewrite the test again. There us definitely some misunderstanding from my side of scenario where this condition occures

Reply:
The scenario (submitting diagnosis keys) only works when the result is positive
So the test is successful only if there are no keys and the error dialog is shown.

1. When the result is positive, that card is visible on the HomeFragment
2. By clicking on that card we go to the test page where user can choose to share the keys
3. then SubmissionResultPositiveOtherWarningFragment where our logic is in place to check if the keys are present.
4. upon no keys, error dialog is shown where the test registers a click on the button
5. Redirection to HomeFragment happens 
6. Test checks if redirection has happened successfully.

Reply:
@Fabian-K Integration test has been added that mimics 0 keys while submitting.